source/partial-deep.d.ts

```diff
@@ -0,0 +1,46 @@
+/**
+Create a type from another type with all properties set to optional and the nested properties of all properties set to optional.
+
+Use cases:
+- Merging a default settings/config object with another object, the second object would be a deep partial of the default object.
+- Mocking and testing complex entities, where populating an entire object with it's properties would be redundant in terms of the mock or test.
+
+@example
+```
+import {PartialDeep} from 'type-fest';
+
+interface Baz {
+    qux: string;
+    corge: {
+        waldo: number[];
+        fred: string;
+    };
+}
+
+function foo(bar: PartialDeep<Baz>) { }
+
+foo({ corge: { waldo: [ 1, 2 ] } });
+```
```

Comment:
This is not really a realistic example. You also don't show the actual usage of `Baz`, only the partial version of it. The point is to show how a type can be useful as both in its original form and as a `PartialDeep` form, in a realistic scenario.

Reply:
I provided a more realistic usage of the type where both a type and it's PartialDeep version are used in c48b576. I think I could use some further feedback here.